### PR TITLE
Multiple fixes on dataprep scripts and train

### PR DESCRIPTION
**Pull request by larroy**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "mxnet"]
 	path = mxnet
-	url = https://github.com/zhreshold/mxnet
+	url = https://github.com/larroy/mxnet.git
```
Comment (zhreshold) [inline .gitmodules]: Why update the submodule?

File: tools/prepare_dataset.py [hunk 2]
```diff
@@ -75,9 +75,9 @@ def parse_args():
 
     print("List file {} generated...".format(args.target))
 
-    subprocess.check_call(["python",
+    subprocess.check_call(["python3",
```
Comment (zhreshold) [inline tools/prepare_dataset.py:79]: python3 may not be always available

File: .gitmodules (hunk 1 above)
Comment (larroy) [reply .gitmodules]: im2rec is too old in your fork.

**Comment by larroy** [comment]
I'm updating your changes to the tip of mxnet.... I will update the PR once is ready

File: tools/prepare_dataset.py (hunk 2 above)
Comment (larroy) [reply tools/prepare_dataset.py:79]: python2 is obsolete

**Comment by larroy** [comment]
I rebased your changes against master in mxnet: 

https://github.com/larroy/mxnet/tree/yolo

**Comment by larroy** [comment]
Are there any other changes needed? I saw additional commits in mshadow

**Comment by larroy** [comment]
I rebase your changes against the latest master, could you have a look? seems to work for me.

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,3 @@
 [submodule "mxnet"]
 	path = mxnet
-	url = https://github.com/zhreshold/mxnet
+	url = https://github.com/larroy/mxnet
```
Comment (zhreshold) [inline .gitmodules:3]: What's the point of update the submodule?

File: custom_mxnet.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Will install a custom mxnet version on py3.5env virtualenv
+# Tested on Ubuntu deep learning AMI https://aws.amazon.com/marketplace/pp/B06VSPXKDX
+set -e
+set -x
+VENV=py3.5env
```
Comment (zhreshold) [inline custom_mxnet.sh:7]: You can rename it as custom_mxnet_py3.sh
However, a python2/3 compatible script is recommended.

File: tools/prepare_dataset.py (hunk 2 above)
Comment (zhreshold) [reply tools/prepare_dataset.py:79]: Let user decide which version to use, python is a link to whatever version.

File: train.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
```
Comment (zhreshold) [inline train.py:1]: Same compatibility issue

File: README.md [hunk 6]
```diff
@@ -29,27 +29,22 @@ python demo.py -h
...
-```
-# cd /path/to/mxnet-ssd
-bash tools/prepare_pascal.sh
-# or if you are using windows
-python tools/prepare_dataset.py --dataset pascal --year 2007,2012 --set trainval --target ./data/train.lst
-python tools/prepare_dataset.py --dataset pascal --year 2007 --set test --target ./data/val.lst --shuffle False
+- Download and prepre the PASCAL VOC dataset.
+```
+./custom_mxnet.sh
+source py3.5env/bin/activate
```
Comment (zhreshold) [inline README.md:35]: virtualenv should be an option rather than forced to do so.

**Review by zhreshold — CHANGES_REQUESTED** [review]

**Comment by zhreshold** [comment]
Overall, I appreciate for the contribution since I don't have much time refactoring this repo recently. However, I would like a more general solution that user can choose the OS/python version, etc. So bash scripts and hard-coded python version should be avoided.

File: train.py (hunk 5 above)
Comment (larroy) [reply train.py:1]: What do you suggest in this case?

File: .gitmodules (hunk 3 above)
Comment (larroy) [reply .gitmodules:3]: I rebased against master, your mxnet subrepo is outdated and have python3 problems that are solved in more recent versions. Of course prior to merging you could also pull my rebase of your changes to mxnet with the Yolo C++ operators so your subrepo is up to date, and this points to your subrepo on the final merge.

**Comment by larroy** [comment]
@zhreshold Do you have any plans to send a PR with the Yolo operators to MXNet master?

**Comment by zhreshold** [comment]
The only reason I didn't send PR is that I cannot yet reproduce the performance of the number provided in paper. And since mxnet is updating so rapidly, it will be very hard to debug into some parts if smth changed. 
In my plan, once the repo is good enough I will send PR to share the operators.

**Comment by zhreshold** [comment]
Okay, here's the solution I think would work for now. I forked a new branch called py3, can you submit a new PR to py3 branch with these changes, so people explicitly require python3 to work now can use that branch?

**Comment by larroy** [comment]
sounds good, I could also make the changes py2 - py3 compat...

**Pull request closed without merging**